po/zh-TW.po

```diff
@@ -5197,6 +5296,137 @@ msgid ""
+"     }\n"
+"     ```\n"
+"\n"
+"    如果沒有 `repr`，判別值類型會需要 2 個位元組，因為 10001 適合 2 個位元"
+"組。\n"
+"\n"
+"\n"
+" * 請嘗試其他類型，例如以下項目：\n"
+" \n"
+" * `dbg_size!(bool)`：大小為 1 個位元組，對齊：1 個位元組。\n"
+" * `dbg_size!(Option<bool>)`：大小為 1 個位元組，對齊：1 個位元組 (區位最佳"
+"化，請見下文)。\n"
+"     * `dbg_size!(&i32)`：大小為 8 個位元組，對齊：8 個位元組 (在 64 位元機器"
+"上)。\n"
+"     * `dbg_size!(Option<&i32>)`：大小為 8 個位元組，對齊：8 個位元組 (空值指"
+"標最佳化，請見下文)。\n"
+"\n"
+" * 區位最佳化：Rust 會為列舉判別值合併未使用的位元模式。\n"
+"\n"
+" * 空值指標最佳化：針對[部分類型](https://doc.rust-lang.org/std/option/"
+"#representation)，Rust 保證 `size_of::<T>()` 等於 `size_of::<Option<T>>()`.\n"
+"\n"
+"     如果想示範位元表示法實際運作時「可能」**的樣子，可以使用下列範例程式碼。"
+"請務必注意，編譯器並無對這個表示法提供保證，因此這完全不安全。\n"
+"\n"
+"     ```rust,editable\n"
+"     use std::mem::transmute;\n"
+"\n"
+"     macro_rules! dbg_bits {\n"
+"         ($e:expr, $bit_type:ty) => {\n"
+"             println!(\"- {}: {:#x}\", stringify!($e), transmute::<_, "
+"$bit_type>($e));\n"
+"         };\n"
+"     }\n"
+"\n"
+"     fn main() {\n"
+"         // TOTALLY UNSAFE. Rust provides no guarantees about the bitwise\n"
+"         // representation of types.\n"
+"         unsafe {\n"
+"             println!(\"Bitwise representation of bool\");\n"
+"             dbg_bits!(false, u8);\n"
+"             dbg_bits!(true, u8);\n"
+"\n"
+"             println!(\"Bitwise representation of Option<bool>\");\n"
+"             dbg_bits!(None::<bool>, u8);\n"
+"             dbg_bits!(Some(false), u8);\n"
+"             dbg_bits!(Some(true), u8);\n"
+"\n"
+"             println!(\"Bitwise representation of Option<Option<bool>>\");\n"
+"             dbg_bits!(Some(Some(false)), u8);\n"
+"             dbg_bits!(Some(Some(true)), u8);\n"
+"             dbg_bits!(Some(None::<bool>), u8);\n"
+"             dbg_bits!(None::<Option<bool>>, u8);\n"
+"\n"
+"             println!(\"Bitwise representation of Option<&i32>\");\n"
+"             dbg_bits!(None::<&i32>, usize);\n"
+"             dbg_bits!(Some(&0i32), usize);\n"
+"         }\n"
+"     }\n"
+"     ```\n"
+"\n"
+"     如果想討論將超過 256 個 `Option` 鏈結在一起的情況，可以使用下列更複雜的"
+"範例。\n"
+"\n"
+"     ```rust,editable\n"
+"     #![recursion_limit = \"1000\"]\n"
+"\n"
+"     use std::mem::transmute;\n"
+"     \n"
+"     macro_rules! dbg_bits {\n"
+"         ($e:expr, $bit_type:ty) => {\n"
+"             println!(\"- {}: {:#x}\", stringify!($e), transmute::<_, "
+"$bit_type>($e));\n"
+"         };\n"
+"     }\n"
+"\n"
+"     // 在 2^n Some() 中包裝值的巨集，n 為「@」符號的數字。\n"
+"     // 評估此巨集時，必須提高遞迴限制。\n"
+"     macro_rules! many_options {\n"
+"         ($value:expr) => { Some($value) };\n"
+"         ($value:expr, @) => {\n"
+"             Some(Some($value))\n"
+"         };\n"
+"         ($value:expr, @ $($more:tt)+) => {\n"
+"             many_options!(many_options!($value, $($more)+), $($more)+)\n"
+"         };\n"
+"     }\n"
+"\n"
+"     fn main() {\n"
+"         // TOTALLY UNSAFE. Rust provides no guarantees about the bitwise\n"
+"         // representation of types.\n"
```

Comment:
What about other comments in code such as this one?

Reply:
Ideally, we should try to minimize the amount of comments in the code. The whole example here is a little too complicated for my taste, so I would like to refactor or remote it completely.